### DISCUSSION TITLE
Empty attributes in CRD Application show always out of sync

### DISCUSSION THREAD

**Discussion by ffppmm** [argoproj/argo-cd#16757] [Ideas]
Hi,

don't know if this is an Issue of ArgoCD,I need a possibility to turn "autosync" explicitly off with:
```yaml
spec:
  syncPolicy: {}
```
So something like:
```yaml
spec:
  syncPolicy:
    manually: true
```

Any thoughts on this?

It would be very nice to have a possibility to explicitly turn off autosync (git "Source of truth"). If somebody turns it on, just for example, I will not see a diff/out of sync because syncPolicy is not "defined" in the source template and so no diff,

Regards Philipp

**Comment by ffppmm** [comment]
sry I need to mention I'm currently testing 2.10 rc1

regards Philipp

**Comment by jannfis** [comment]
Often times, when you specify a default value in a resource, the resource will be normalized before it's stored.

Since an empty sync policy (e.g. `{}`) is the default, the resource as returned by the API server will omit that field, leading to an out of sync condition.

Your problem/use case is valid though, and I believe there are some enhancement requests for what you describe in our tracker already.

IMO, a good approach would be an `enabled` flag within `.spec.syncPolicy.automated`, which would default to `true`. Basically, it could be turned off in Git like the follows:

```yaml
spec:
  syncPolicy:
     automated:
        enabled: false
```

**Reply by ffppmm** [reply]
Hi,

I would not default to true or false, if possible, because it would change "standard behavior". Imho it would be very convenient to have the possibility to explicitly turn it off as shown in your code snippet. 

Just my thoughts,
Regards Philipp
